### PR TITLE
Codespell

### PR DESCRIPTION
**Pull request by croghostrider**
## Description

add codespell to pre-commit

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code quality improvements to existing code or addition of tests

## Checklist

- [ ] The documentation has been adjusted accordingly
- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] The changes are documented in the changelog (docs/changelog.md)

### CODE REVIEW

File: requirements/testing.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ pytest-cov==4.0.0
 setuptools==67.1.0
 tox==4.4.4
 mypy==0.991
+codespell==2.2.2
```
Comment (farmio) [inline requirements/testing.txt:14]: Do we need this as explicit requirement when we only use it in per-commit?

File: xknx/io/tunnel.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ async def _reconnect(self) -> None:
             self.transport.stop()
         await asyncio.sleep(self.auto_reconnect_wait)
         if await self.connect():
-            logger.info("Successfully reconnected to KNX bus.")
+            logger.info("successfully reconnected to KNX bus.")
```
Comment (farmio) [inline xknx/io/tunnel.py:162]:
```suggestion
            logger.info("Successfully reconnected to KNX bus.")
```
Why is this changed to lower case?

**Review by farmio — CHANGES_REQUESTED** [review]

**Comment by farmio** [comment]
I was planning to add `ruff` to per-commit. Do you happen to know if codespell linting can be done from ruff too?

File: requirements/testing.txt (hunk 1 above)
Comment (croghostrider) [reply requirements/testing.txt:14]: I wasn't sure if you track this way if the pre-commit requirements are up to date.

**Comment by croghostrider** [comment]
> I was planning to add `ruff` to per-commit. Do you happen to know if codespell linting can be done from ruff too?

I don't believe.

**Comment by farmio** [comment]
https://github.com/XKNX/xknx/actions/runs/4105036064/jobs/7081339573
It is installed, but not run. You'll need to add it to [`tox.ini`](https://github.com/XKNX/xknx/blob/e133b775abdff5e2f31f9ba8715c472b9dc3f696/tox.ini#L14)

**Comment by farmio** [comment]
When you don't add it to `[testenv:lint]` but create a new env, you'll have to add that to `envlist` in tox.ini.
Personally I'd add it to the `lint` environment.

**Review by farmio — APPROVED** [review]
Thank you 👍
